base/linalg/special.jl

```diff
@@ -52,12 +64,16 @@ function convert(::Type{Bidiagonal}, A::Tridiagonal)
 end
 
 function convert(::Type{SymTridiagonal}, A::Tridiagonal)
-    all(A.dl .== A.du) || throw(ArgumentError("Matrix cannot be represented as SymTridiagonal"))
+    if !all(A.dl .== A.du)
+        throw(ArgumentError("Matrix cannot be represented as SymTridiagonal"))
+    end
     SymTridiagonal(A.d, A.dl)
 end
 
 function convert(::Type{Diagonal}, A::AbstractTriangular)
-    full(A) == diagm(diag(A)) || throw(ArgumentError("Matrix cannot be represented as Diagonal"))
+    if full(A) != diagm(diag(A))
```

Comment:
Yes. That is bananas. Allocating three arrays for an operation that doesn't require a single is not efficient. I'll prepare a fix when this is merged.